packages/cardpay-sdk/sdk/utils/general-utils.ts

```diff
@@ -312,6 +314,18 @@ export async function sendTransaction(web3: Web3, transaction: Transaction): Pro
   return txHash;
 }
 
+export function extractSendTransactionError(revert: Revert, abiInterface: Interface) {
+  let error;
+  try {
+    let funcError = abiInterface.parseError(revert.data);
+    error = `${funcError.name} ${funcError.args.join(',')}`;
```

Comment:
I think it would be great to document all the types of errors/reasons that we know that could happen here. This way we can handle them in the hub, for example if gas limit is reached we can try again later, or if the sp is unknown or the time period is invalid, we can stop retrying.